### PR TITLE
Return StatusOr for Table::Apply()

### LINKED ISSUE #1969: Table::Apply() needs to be changed to use StatusOr or Status for void
void Table::Apply(SingleRowMutation&& mut)

Files affected:

/bigtable/examples/data_snippets.cc
void Apply()

/bigtable/examples/bigtable_hello_world.cc

/bigtable/examples/bigtable_hello_app_profile.cc
/bigtable/benchmarks/endurance_benchmark.cc
/bigtable/benchmarks/apply_read_latency_benchmark.cc
/bigtable/benchmarks/embedded_server_test.cc

/bigtable/tests/data_integration_test.cc
void DataIntegrationTest::Apply
TEST_F(DataIntegrationTest, TableApply)

/bigtable/table_apply_test
TEST_F(TableApplyTest, Simple)
TEST_F(TableApplyTest, Failure)
TEST_F(TableApplyTest, Retry)
TEST_F(TableApplyTest, RetryIdempotent)

/bigtable/tests/mutations_integration_test.cc
TEST_F(MutationIntegrationTest, DeleteFromColumnForTimestampRangeTest)
TEST_F(MutationIntegrationTest, DeleteFromColumnForReversedTimestampRangeTest)
TEST_F(MutationIntegrationTest, DeleteFromColumnForAllTest)
TEST_F(MutationIntegrationTest, DeleteFromColumnForEmptyTimestampRangeTest)
TEST_F(MutationIntegrationTest, DeleteFromColumnStartingFromTest)
TEST_F(MutationIntegrationTest, DeleteFromColumnEndingAtTest) 
TEST_F(MutationIntegrationTest, DeleteFromFamilyTest) 
TEST_F(MutationIntegrationTest, DeleteFromRowTest)

### PR DESCRIPTION
**Pull request by rsimion**
Fixes #1969

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2044)

### CODE REVIEW

File: google/cloud/bigtable/table.cc [hunk 1]
```diff
@@ -48,12 +48,19 @@ inline namespace BIGTABLE_CLIENT_NS {
...
               "bigtable::Table must be CopyAssignable");
 
-void Table::Apply(SingleRowMutation&& mut) {
+StatusOr<std::vector<FailedMutation>> Table::Apply(SingleRowMutation&& mut) {
   std::vector<FailedMutation> failures = impl_.Apply(std::move(mut));
   if (!failures.empty()) {
-    grpc::Status status = failures.front().status();
-    ReportPermanentFailures(status.error_message().c_str(), status, failures);
+    // grpc::Status status = failures.front().status();
+    // if (!status.ok()) {
```
Comment (remyabel) [inline google/cloud/bigtable/table.cc]: Remove commented code?

Comment (rsimion) [reply google/cloud/bigtable/table.cc]: I will, just wanted @coryan to see if it makes sense.

Comment (rsimion) [reply google/cloud/bigtable/table.cc]: Removed it.

**Comment by rsimion** [comment]
@coryan 
I am not sure how to change 2 functions in table_apply_test.cc :
TEST_F(TableApplyTest, Failure)
and
TEST_F(TableApplyTest, RetryIdempotent)
That is why I commented out for now.

**Comment by rsimion** [comment]
@coryan 
Those two functions(see below) in mutations_integration_test.cc do not use the emulator.
I output both status.ok() and status and after I run ./mutations_integration_test.cc <project_id> <instance_id> I get:

[ RUN      ] MutationIntegrationTest.DeleteFromColumnForReversedTimestampRangeTest
status.ok(): 0
status: Error in field 'Mutation list' : Error in element #0 : Inverted or empty timestamp range [4000, 2000) [INVALID_ARGUMENT]
[       OK ] MutationIntegrationTest.DeleteFromColumnForReversedTimestampRangeTest (2406 ms)
[ RUN      ] MutationIntegrationTest.DeleteFromColumnForEmptyTimestampRangeTest
status.ok(): 0
status: Error in field 'Mutation list' : Error in element #0 : Inverted or empty timestamp range [2000, 2000) [INVALID_ARGUMENT]
[       OK ] MutationIntegrationTest.DeleteFromColumnForEmptyTimestampRangeTest (2300 ms)

 I used EXPECT_FALSE(status.ok()) for both instead of ASSERT_STATUS_OK(status).

**Pull request merged**